last release (21.1.0) sends a lot of errors

Throwing random errors

```python
[2023-07-18 13:04:10 +0000] [3150] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/gunicorn/arbiter.py", line 609, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 95, in init_process
    super().init_process()
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/base.py", line 142, in init_process
    self.run()
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 232, in run
    self.murder_keepalived()
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 176, in murder_keepalived
    self.poller.unregister(conn.sock)
  File "/usr/local/lib/python3.11/selectors.py", line 366, in unregister
    key = super().unregister(fileobj)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/selectors.py", line 249, in unregister
    key = self._fd_to_key.pop(self._fileobj_lookup(fileobj))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/selectors.py", line 225, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/selectors.py", line 42, in _fileobj_to_fd
    raise ValueError("Invalid file descriptor: {}".format(fd))
ValueError: Invalid file descriptor: -1
[2023-07-18 13:04:10 +0000] [3150] [INFO] Worker exiting (pid: 3150)
[2023-07-18 13:04:11 +0000] [1] [ERROR] Worker (pid:3150) exited with code 255
[2023-07-18 13:04:11 +0000] [1] [ERROR] Worker (pid:3150) exited with code 255.
```

The previous versions did not have this issue at all, was triggered in our latest build, so we’ll rollback version but still the problem remains in your latest release.

Our CI (shame on us) installed 21.1.0 at 10:45 EST this am, upgrading us from 20.1.0, and we get tons of the following, causing 502's to return from nginx (sitting in front of gunicorn)

```
│ overwatch-api-green-67c57565dd-fg225 api [2023-07-18 17:33:38 +0000] [55] [ERROR] Socket error processing request.                                                                                                         │
│ overwatch-api-green-67c57565dd-fg225 api Traceback (most recent call last):                                                                                                                                                │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/workers/gthread.py", line 285, in handle                                                                   │
│ overwatch-api-green-67c57565dd-fg225 api     keepalive = self.handle_request(req, conn)                                                                                                                                    │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/workers/gthread.py", line 357, in handle_request                                                           │
│ overwatch-api-green-67c57565dd-fg225 api     util.reraise(*sys.exc_info())                                                                                                                                                 │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/util.py", line 641, in reraise                                                                             │
│ overwatch-api-green-67c57565dd-fg225 api     raise value                                                                                                                                                                   │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/workers/gthread.py", line 343, in handle_request                                                           │
│ overwatch-api-green-67c57565dd-fg225 api     resp.write(item)                                                                                                                                                              │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 326, in write                                                                          │
│ overwatch-api-green-67c57565dd-fg225 api     self.send_headers()                                                                                                                                                           │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/http/wsgi.py", line 322, in send_headers                                                                   │
│ overwatch-api-green-67c57565dd-fg225 api     util.write(self.sock, util.to_bytestring(header_str, "latin-1"))                                                                                                              │
│ overwatch-api-green-67c57565dd-fg225 api   File "/root/.pyenv/versions/3.8/lib/python3.8/site-packages/gunicorn/util.py", line 299, in write                                                                               │
│ overwatch-api-green-67c57565dd-fg225 api     sock.sendall(data)                                                                                                                                                            │
│ overwatch-api-green-67c57565dd-fg225 api OSError: [Errno 9] Bad file descriptor 
```

--- a/gunicorn/workers/gthread.py
+++ b/gunicorn/workers/gthread.py
@@ -122,13 +122,10 @@ class ThreadWorker(base.Worker):
             sock, client = listener.accept()
             # initialize the connection object
             conn = TConn(self.cfg, sock, client, server)
-            # set timeout to ensure it will not be in the loop too long
-            conn.set_timeout()
 
             self.nr_conns += 1
             # wait until socket is readable
             with self._lock:
-                self._keep.append(conn)
                 self.poller.register(conn.sock, selectors.EVENT_READ,
                                      partial(self.on_client_socket_readable, conn))
         except EnvironmentError as e:

